### ISSUE TITLE
Automatic setting of MACHINE does not work?

### ISSUE THREAD

**Issue by nuclearfusionreactor** [PrincetonUniversity/STELLOPT#150]
From the `README.md` file

```
... The `make.inc` file
in the main directory is a script which picks these files based on what
is returned by the `uname -n` command on your machine.  This can be
overridden by setting the environment variable `MACHINE` to the file you
wish to use. ...
```
I would expect, if I add a file `make_machine.inc`, where `machine` is what `uname -n` returns, in `SHARE`, and then run `./build_all` that then this settings file is used.

This appears not the be the case. Instead it seems `MACHINE` in `make.inc` is not set (and thus no file included), unless I explicitly set it on the commandline, e.g. `MACHINE=machine ./build_all`.

It seems to me, that the `awk` command to parse the result of `uname -n` returns nothing, if no match is found.
If this is the intended behaviour, then I would suggest to modify `README.md` to reflect this (i.e. to state that you either have to use the call with `MACHINE=...` or that you have to modify `make.inc`.
As an alternative, and I think more user friendly approach, I would suggest to change the awk command.
A possible solution is attached as a patch file (I had to add `txt` as extension as I got the message that `patch` is not recognized).
I just read some tutorials on `awk`, so I am quite sure it is not the best solution. Nevertheless, it should demonstrate what I think is the better approach: check the result of `uname -n` for various possibilities, with those that should give the same result in one (else)if clause, and if nothing matches, return the result unchanged. This removes the requirement for some of the previous cases.

There could be a drawback to the proposed approach (I am not familiar enough with `awk` to be sure). I do not know if it is possible to have regular expressions in the (else)if-conditions. To give an example why this could be important: the machines here have names of the form `some name+two digit number`. As the location of the libraries should be the same on all machines, a single configuration file should be enough, thus one would like to have a (else)if-clause that filters out all these names and then returns `some name`, which should be easy with regular expressions.

[make.inc.patch.txt](https://github.com/PrincetonUniversity/STELLOPT/files/8378292/make.inc.patch.txt)

PS: I noticed that in the current approach more than one case can be executed. It seems me, that this is only relevant if some case is repeated. I am not sure if this is an advantage or a disadvantage of the current or the proposed approach.

**Comment by zhucaoxiang** [comment]
@nuclearfusionreactor Thanks for creating this issue. This `make.inc` was mainly developed by @lazersos . Let me try to summarize and answer your points.
1. `awk` returns nothing if no match is found.
     I don't think this is a flaw. Returning an unmatched `uname` will also not work because you cannot find the corresponding `make_$(MACHINE).inc` file. 
2. Your attachment.
     I think you cannot use `==` because in many cases, the uname contains other information. For example, my user name on the PPPL portal cluster was `czhu@sunfirexXX.pppl.gov`, where `XX` was the login node number. 

Ovearall, I think you are write the current configuration system and documentation can be improved. If you have furthur thoughts, please share with us. Or you can directly create a pull request.

**Comment by nuclearfusionreactor** [comment]
1. If I understand you correctly, then maybe we have different use cases in mind. I was thinking of simply creating a file `make_$MACHINE).inc` with the exact output of `uname -n`. While i guess you were thinking of creating a file were `MACHINE` would be the common part among the local group, which of course would require to change the output of `uname -n` in the script - what is done with `awk`.

2. While `uname -n` is not about the username, you are still right, the nodename might only contain the strings I compare to.
So I have to change what I proposed, because it is not actually a solution of the problem.

Overall, I think now, this is not a bug but intended behaviour, while the documentation makes this not clear.

Edit:
I reviewed my approach, and changed it to make use of regular expressions (~) instead of direct comparison (==).

[make.inc.patch.txt](https://github.com/PrincetonUniversity/STELLOPT/files/8381589/make.inc.patch.txt)

To my understanding it should work as the current version in the cases of the explicitly given machines. It differs in the behaviour for machines not given explicitly, as it will return `uname -n` unchanged in this case.

If this behaviour is not desired, then I would suggest to change `README.md`. To give a rough example:
```
...
`MACHINE=bobdole` before calling the build scripts.
To add a new machine configuration file, first check `uname -n` for your system,
then add an entry in the `awk` part of `make.inc`. 
Once this is done, create a corresponding `make_$MACHINE.inc` in the `SHARE` folder.
It is also important to set `STELLOPT_PATH` to the path to your current directory
where you've pulled the repository.
...
```
